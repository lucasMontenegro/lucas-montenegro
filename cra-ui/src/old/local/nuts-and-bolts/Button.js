import React from 'react';
import styled from 'styled-components';
import palette from './palette';

const createButton = color => styled.button`
  margin: 0 0.5ch;
  border: 0;
  padding: 0.75em 2ch;
  border-radius: 2px;
  color: ${palette[color].contrast}
  background-color: ${palette[color].normal};

  :hover {
    background-image: linear-gradient(transparent,rgba(0,0,0,.05) 40%,rgba(0,0,0,.1));
  }

  :disabled {
    opacity: 0.4;
  }

  :active {
    box-shadow: 0 0 0 1px rgba(0,0,0,.15) inset, 0 0 6px rgba(0,0,0,.2) inset;
  }
`;

const GrayButton = createButton('gray');
const BlueButton = createButton('blue');
const RedButton = createButton('red');
const GreenButton = createButton('green');
const OrangeButton = createButton('orange');

const Button = ({ primary, error, success, warning, children, ...props }) => {
  return primary ? <BlueButton {...props}>{children}</BlueButton>
  : error ? <RedButton {...props}>{children}</RedButton>
  : success ? <GreenButton {...props}>{children}</GreenButton>
  : warning ? <OrangeButton {...props}>{children}</OrangeButton>
  : <GrayButton  {...props}>{children}</GrayButton>
}

export default Button;
