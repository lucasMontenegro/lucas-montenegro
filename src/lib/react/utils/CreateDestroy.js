import React, { useState } from "react"
import Div from "lib/react/utils/Div"
import Button from "@material-ui/core/Button"
export default function CreateDestroy ({ id, box, Component, props }) {
  const [render, setRender] = useState(false)
  return (
    <Div box={box} color="blue">
      <Div>
        <Button
          id={id}
          variant="outlined"
          color="primary"
          onClick={() => setRender(render => !render)}
        >
          CREATE/DESTROY
        </Button>
      </Div>
      <Div><em>{render ? `alive` : `dead`}</em></Div>
      {render && <Component {...(props || {})} />}
    </Div>
  )
}
