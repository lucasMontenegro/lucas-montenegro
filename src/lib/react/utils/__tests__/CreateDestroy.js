import React, { useState } from "react"
import Button from "@material-ui/core/Button"
import renderer from "react-test-renderer"
import CreateDestroy from "../CreateDestroy.js"
jest.mock(`react`, () => {
  const React = jest.requireActual("react")
  return {
    __esModule: true,
    ...React,
    default: React,
    useState: jest.fn(),
  }
})
jest.mock(`lib/react/utils/Div`, () => {
  const React = jest.requireActual("react")
  return { __esModule: true, default: props => <div {...props} className="Div" /> }
})
jest.mock(`@material-ui/core/Button`, () => {
  const React = jest.requireActual("react")
  return { __esModule: true, default: jest.fn(props => <button {...props} />) }
})
describe(`../index.js`, () => {
  it(`should use the right dependency versions`, () => {
    expect(jestUtils.getDependencies([`react`, `@material-ui/core`])).toMatchSnapshot()
  })
  const cases = [
    [true, { children: `foo` }, false],
    [true, null, false],
    [false, { children: `foo` }, true],
    [false, null, true],
  ]
  const msg = `<CreateDestroy />  (render %j, Component props %j)`
  describe.each(cases)(msg, (render, props, newRender) => {
    let html
    const setRender = jest.fn()
    beforeAll(() => {
      useState.mockReturnValueOnce([render, setRender])
      function Component (props) {
        return <div className="Component">{props.children}</div>
      }
      html = renderer.create(
        <CreateDestroy id="id" box={256} Component={Component} props={props} />
      )
    })
    it(`should render`, () => {
      expect(html.toJSON()).toMatchSnapshot()
    })
    describe(`Button props.onClick`, () => {
      let setRenderCallback
      beforeAll(() => {
        try {
          Button.mock.calls[0][0].onClick()
          Button.mockClear()
          setRenderCallback = setRender.mock.calls[0][0]
        } catch (e) {}
      })
      it(`should call setRender`, () => {
        expect(setRender.mock.calls).toHaveLength(1)
        expect(setRender.mock.calls[0]).toHaveLength(1)
        expect(setRender.mock.calls[0][0]).toBeInstanceOf(Function)
      })
      describe(`setRender callback`, () => {
        it(`should toggle the render value`, () => {
          expect(setRenderCallback(render)).toBe(newRender)
        })
      })
    })
  })
})