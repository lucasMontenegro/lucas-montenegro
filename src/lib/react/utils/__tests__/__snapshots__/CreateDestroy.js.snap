// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`../index.js <CreateDestroy />  (render false, Component props {"children":"foo"}) should render 1`] = `
<div
  box={256}
  className="Div"
  color="blue"
>
  <div
    className="Div"
  >
    <button
      color="primary"
      id="id"
      onClick={[Function]}
      variant="outlined"
    >
      CREATE/DESTROY
    </button>
  </div>
  <div
    className="Div"
  >
    <em>
      dead
    </em>
  </div>
</div>
`;

exports[`../index.js <CreateDestroy />  (render false, Component props null) should render 1`] = `
<div
  box={256}
  className="Div"
  color="blue"
>
  <div
    className="Div"
  >
    <button
      color="primary"
      id="id"
      onClick={[Function]}
      variant="outlined"
    >
      CREATE/DESTROY
    </button>
  </div>
  <div
    className="Div"
  >
    <em>
      dead
    </em>
  </div>
</div>
`;

exports[`../index.js <CreateDestroy />  (render true, Component props {"children":"foo"}) should render 1`] = `
<div
  box={256}
  className="Div"
  color="blue"
>
  <div
    className="Div"
  >
    <button
      color="primary"
      id="id"
      onClick={[Function]}
      variant="outlined"
    >
      CREATE/DESTROY
    </button>
  </div>
  <div
    className="Div"
  >
    <em>
      alive
    </em>
  </div>
  <div
    className="Component"
  >
    foo
  </div>
</div>
`;

exports[`../index.js <CreateDestroy />  (render true, Component props null) should render 1`] = `
<div
  box={256}
  className="Div"
  color="blue"
>
  <div
    className="Div"
  >
    <button
      color="primary"
      id="id"
      onClick={[Function]}
      variant="outlined"
    >
      CREATE/DESTROY
    </button>
  </div>
  <div
    className="Div"
  >
    <em>
      alive
    </em>
  </div>
  <div
    className="Component"
  />
</div>
`;

exports[`../index.js should use the right dependency versions 1`] = `
Object {
  "@material-ui/core": "^4.5.0",
  "react": "^16.10.1",
}
`;
